presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestMaterializedViewQueryOptimizer.java

```diff
@@ -464,436 +482,871 @@ public void testWithJoinTables()
+        assertOptimizedQuery(baseQuerySql, baseQuerySql, ImmutableMap.of(BASE_TABLE_1, ImmutableList.of(originalViewSql)));
 
         originalViewSql = format("SELECT a, b, c FROM %s WHERE a IN (5,6)", BASE_TABLE_1);
         baseQuerySql = format("SELECT a, b, c FROM %s WHERE a = 7", BASE_TABLE_1);
 
-        assertOptimizedQuery(originalViewSql, baseQuerySql, baseQuerySql);
+        assertOptimizedQuery(baseQuerySql, baseQuerySql, ImmutableMap.of(BASE_TABLE_1, ImmutableList.of(originalViewSql)));
 
         originalViewSql = format("SELECT a, b, c FROM %s WHERE a NOT IN (5,6)", BASE_TABLE_1);
         baseQuerySql = format("SELECT a, b, c FROM %s WHERE a <= 5", BASE_TABLE_1);
 
-        assertOptimizedQuery(originalViewSql, baseQuerySql, baseQuerySql);
+        assertOptimizedQuery(baseQuerySql, baseQuerySql, ImmutableMap.of(BASE_TABLE_1, ImmutableList.of(originalViewSql)));
 
         originalViewSql = format("SELECT a, b, c FROM %s WHERE a NOT IN (5,6)", BASE_TABLE_1);
         baseQuerySql = format("SELECT a, b, c FROM %s WHERE a NOT IN (6,7)", BASE_TABLE_1);
 
-        assertOptimizedQuery(originalViewSql, baseQuerySql, baseQuerySql);
+        assertOptimizedQuery(baseQuerySql, baseQuerySql, ImmutableMap.of(BASE_TABLE_1, ImmutableList.of(originalViewSql)));
 
         originalViewSql = format("SELECT a, b, c FROM %s WHERE a NOT IN (5,6)", BASE_TABLE_1);
         baseQuerySql = format("SELECT a, b, c FROM %s WHERE a IN (6,7)", BASE_TABLE_1);
 
-        assertOptimizedQuery(originalViewSql, baseQuerySql, baseQuerySql);
+        assertOptimizedQuery(baseQuerySql, baseQuerySql, ImmutableMap.of(BASE_TABLE_1, ImmutableList.of(originalViewSql)));
+    }
+
+    @Test
+    public void singleSubquerySingleCompatibleView()
+    {
+        String subquery = format("SELECT a, b FROM %s", BASE_TABLE_1);
+        String originalViewSql = subquery;
+
+        String baseQuerySql = format("SELECT a, b FROM (%s)", subquery);
+        String expectedRewrittenSql = format("SELECT a, b FROM (SELECT a, b FROM (%s))", VIEW_1);
+
+        assertOptimizedQuery(baseQuerySql, expectedRewrittenSql, ImmutableMap.of(BASE_TABLE_1, ImmutableList.of(originalViewSql)));
+    }
+
+    @Test
+    public void singleSubquerySingleIncompatibleView()
+    {
+        String subquery = format("SELECT a, b, c FROM %s", BASE_TABLE_1);
+        String originalViewSql = format("SELECT a, b FROM %s", BASE_TABLE_1);
+
+        String baseQuerySql = format("SELECT a, b FROM (%s)", subquery);
+
+        assertOptimizedQuery(baseQuerySql, baseQuerySql, ImmutableMap.of(BASE_TABLE_1, ImmutableList.of(originalViewSql)));
+    }
+
+    @Test
+    public void multipleViewsSameBaseTableAllCompatible()
+    {
+        String subquery1 = format("SELECT a, b FROM %s", BASE_TABLE_1);
+        String subquery2 = format("SELECT c FROM %s", BASE_TABLE_1);
+        String viewSql1 = subquery1;
+        String viewSql2 = subquery2;
+
```

Comment:
Redundant variables. And there are a few other instances in this file as well.

Reply:
The purpose of these is (an attempt) to increase readability. Since I want to re-use the string, but there is a big logical difference between the sql which comprises the MV's definition and the subquery, I decided to instantiate `viewSql` even when it's identical to `subquery`, for the sake of being able to use them where you'd expect them to be- regardless of their equivalence. e.g. registering `subquery` in some of the maps and `viewSql` in others feels like it may hurt readability more than having some redundant assignments. WDYT?